Sources/XCUITestHelpers/XCUIElement+Visibility.swift

```diff
@@ -0,0 +1,10 @@
+import XCTest
+
+public extension XCUIElement {
+
+    func isFullyVisibleOnScreen(in app: XCUIApplication = XCUIApplication()) -> Bool {
+        guard exists && frame.isEmpty == false && isHittable else { return false }
+
+        return app.windows.element(boundBy: 0).frame.contains(frame)
```

Comment:
I must admit, I spent more time than I'm comfortable confessing until I realized that the first `frame` and the second one are different frames.

Also, it's a bit mind-blowing to know that the rect for the `window` corresponds to the phone device screen size, however, hierarchically it still contains the elements that are located far beyond the screen bounds (I guess this is the case when the term 'window' comes really descriptive and handy). Additionally, that the X coordinate for StaticTexts (from the test app) is zero when they are not visible, but it becomes 20 when they become visible.